core/src/main/java/org/elasticsearch/search/rescore/QueryRescorer.java

```diff
@@ -35,15 +36,9 @@
 public final class QueryRescorer implements Rescorer {
 
     public static final Rescorer INSTANCE = new QueryRescorer();
-    public static final String NAME = "query";
 
     @Override
-    public String name() {
-        return NAME;
-    }
-
-    @Override
-    public TopDocs rescore(TopDocs topDocs, SearchContext context, RescoreSearchContext rescoreContext) throws IOException {
+    public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext rescoreContext) throws IOException {
```

Comment:
I agree that exposing the SearchContext here is too sensitive but without it the rescorer cannot access the original query for instance. Maybe replace it with the `SearchSourceBuilder` ?